utils/generate_task_parameters.py

```diff
@@ -0,0 +1,528 @@
+# coding=utf-8
+# Copyright 2024-present, the HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""**Experimental** Utility script to check and update the InferenceClient task methods arguments and docstrings based on the tasks input parameters."""
+
+import argparse
+import builtins
+import inspect
+import re
+import textwrap
+from collections import defaultdict
+from pathlib import Path
+from typing import Dict, List, NoReturn, Optional, Set
+
+import libcst as cst
+from helpers import format_source_code
+from libcst.codemod import CodemodContext
```

Comment:
let's add `libcst` as a dependency in the `"quality"` extra deps here: https://github.com/huggingface/huggingface_hub/blob/ce38f468ae83e4a1e8813e9e72f6906ce614f9bd/setup.py#L95